libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/map/longpress/RoutePreviewLongPressMapComponent.kt

```diff
@@ -35,18 +33,14 @@ internal class RoutePreviewLongPressMapComponent(
         super.onDetached(mapboxNavigation)
         mapView.gestures.removeOnMapLongClickListener(longClickListener)
         hapticFeedback = null
-        this.mapboxNavigation = null
     }
 
     private val longClickListener = OnMapLongClickListener { point ->
-        val mapboxNavigation = mapboxNavigation ?: return@OnMapLongClickListener false
         if (context.options.enableMapLongClickIntercept.value) {
             val location = store.state.value.location?.enhancedLocation
             location?.toPoint()?.also { lastPoint ->
                 store.dispatch(DestinationAction.SetDestination(Destination(point)))
-                val options =
-                    context.routeOptionsProvider.getOptions(mapboxNavigation, lastPoint, point)
-                store.dispatch(RoutePreviewAction.FetchOptions(options))
+                store.dispatch(RoutePreviewAction.FetchRoute)
```

Comment:
Since the `lastPoint` is not used anymore to `getOptions` we can change the if block to be
```
if (context.options.enableMapLongClickIntercept.value) {
    val location = store.state.value.location?.enhancedLocation
    location?.let {
        store.dispatch(DestinationAction.SetDestination(Destination(point)))
        store.dispatch(RoutePreviewAction.FetchRoute)
        hapticFeedback?.tick()
    } ?: logW(TAG, "Current location is unknown so map long press does nothing")
}
```

Reply:
Updated